socs/agents/acu/agent.py

```diff
@@ -401,18 +436,20 @@ def monitor(self, session, params):
                 yield dsleep(1)
                 continue
             for k, v in session.data.items():
-                if isinstance(v, dict):
-                    for (key, value) in v.items():
-                        for category in self.monitor_fields:
-                            if key in self.monitor_fields[category]:
-                                if isinstance(value, bool):
-                                    self.data['status'][category][self.monitor_fields[category][key]] = int(value)
-                                elif isinstance(value, int) or isinstance(value, float):
-                                    self.data['status'][category][self.monitor_fields[category][key]] = value
-                                elif value is None:
-                                    self.data['status'][category][self.monitor_fields[category][key]] = float('nan')
-                                else:
-                                    self.data['status'][category][self.monitor_fields[category][key]] = str(value)
+                if k in not_data_keys:
```

Comment:
Worth still checking if `instance(v, dict)`? The current way this is seems to assume all non-dicts are added to `not_data_keys`.

Reply:
I think it's simpler to enforce the use of not_data_keys for all non-data keys.  If that list is kept up to date, no non-dicts will get through to the loop over .items().